dat/missions/dvaered/dvaered-colonel-transport.lua

```diff
@@ -0,0 +1,127 @@
+
+local dest_planet = spob.getS("Adham")
+
+
+local npc_name = _("Dvaered Colonel")
+local npc_portrait, npc_image
+function create()
+   mem.npc_image, mem.npc_portrait = vni.dvaeredMilitary()
+   misn.setNPC(npc_name, npc_portrait, _("A Dvaered, very professional-looking, is sitting with an excellant posture at the bar.") )
+end
+
+function accept()
+   local accepted = false
+   vn.clear()
+   vn.scene()
+   local m = vn.newCharacter( npc_name {image=npc_image} )
+   vn.transition()
+   m(fmt.f([[As you approach, the Dvaered soldier stands. "Hello, captain {playername}!" they say. "Nice to see you around here. How's it going?"]]))
+     {playername=player.name()}
+   vn.menu{
+      {_([["Quite well, thank you!"]]), "well"},
+      {_([["I guess it's going alright."]]), "fine"},
+   }
+
+   vn.label("well")
+   m(_([["Wonderful! Glad you're having a good time."]]))
+   vn.jump("mission description")
+
+   vn.label("fine")
+   m(_([["As long as there's no trouble..."]]))
+   vn.jump("mission description")
+
+   vn.label("mission description")
+   m(fmt.f([["Well, to the point. I need somebody to escort me and my Arsenal to {pnt}. Would you be willing to do that? I can't tell you why."]]))
+     {pnt=mem.dest_planet}
+   vn.menu{
+      {_([["Remind me what system that's in?"]]), "what system"},
+      {_([["I'd be happy to do that!"]]), "sure"},
+      {_([["What is your name?"]]), "what is your name"},
+   }
+
+   vn.label("what system")
+   m(fmt.f([["Umm..." the Dvaered says. They consult a watch. "Alright, {pnt} is in {sys}."]]))
+     {pnt=mem.dest_planet, sys=mem.dest_sys}
+   vn.jump("choice")
+
+   vn.label("what is your name")
+   m(_([[The Dvaered seems slightly taken aback. "Well... that may be classified information... call me Radver."]]))
+   vn.jump("choice")
+
+   vn.label("sure")
+   m(_([["Wonderful! I'll be on your ship when you leave."]]))
+   vn.func( function ()
+      accepted = true
+   end )
+
+   vn.run()
+
+   vn.label("choice")
+   m(_([["Well? Will you do this?"]]))
+   vn.menu{
+      {_([["Yep, I'd be glad to!"]]), "sure"},
+      {_([["Not going to happen, sorry."]]), "never"},
+   }
+
+   vn.label("never")
+   m(_([["That's sad. Oh well, I'll ask someone else." The Dvaered leaves.]]))
+   vn.done()
+
+   if not accepted then return end
+
+   misn.accept()
+   
+   misn.setReward(750000)
+   misn.setDesc(fmt.f(_("Escort a Dvaered colonel, who is flying an Arsenal, to {pnt} in the {sys} system. You haven't been told why, but there may be a large payment."), {pnt=mem.dest_planet, sys=mem.dest_sys}))
+   misn.osdCreate(_("Dvaered colonel escort"), {
+      fmt.f(_("Escort a Dvaered colonel to {pnt} in the {sys} system.")), {pnt=mem.dest_planet, sys=mem.dest_sys},
+   })
+   misn.markerAdd( mem.dest_planet )
+   hook.land( "land" )
+   hook.enter( "pirate_ambush" )
+   local colonel_ship = ship.get("Dvaered Arsenal")
+   escort.init ( colonel_ship, {
+   })
+end
+
+function pirate_ambush ()
```

Comment:
This is a bit technical, but you should make sure the system isn't claimed exclusively when spawning the pirates. The following line at the top of the enter wook should work.
```lua
if not naev.claimTest( system.cur(), true ) then return end
```
 Is there a reason it's pirates after them? It might make more sense to have it be another Dvaered warlord.

Reply:
Is there a way to do that?
I may not be understanding the whole situation, but it looks like there are no hostile Dvaered governments.
Or were you thinking have a hostile-to-the-Dvaered government flying a Dvaered ship?

Reply:
Maybe the Dreamer Clan?

Reply:
You can do that with dynamic factions. See the dvaered warlord battle event for an example:
https://github.com/naev/naev/blob/main/dat/events/dvaered/warlords_battle.lua

> Maybe the Dreamer Clan?

FLF would probably make more sense. However, it depends on what backstory you want to give the character. I think the most "neutral" would be another Dvaered Warlord. Just create two dynamic factions: one for the colonel, and one for the enemies. Have them be hostile to each other (Colonel should also be hostile to pirates), and it should work.

Reply:
I'm afraid I can't figure out how to set the ship's governments to the dynamic; could you help with that?